File: WebPPTGame/public_html/model/DataContainer.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
/*
var imported = document.createElement('script');
imported.src = '/model/EnumFichas3';
document.head.appendChild(imported);

var imported2 = document.createElement('script');
imported2.src = '/model/EnumFichas5';
document.head.appendChild(imported2);

var imported3 = document.createElement('script');
imported3.src = '/model/EnumFichas9';
document.head.appendChild(imported3);
*/
DataContainer = function(){
    var roundsCounter;
    var roundsLimit;
    var victoriesP1;
    var victoriesP2;
    var turno;
    var modalidadJuego=new ModalidadJuego();
    var enumChosen1;
    var enumChosen2;;
    var factorAlgoritmo;
    var idImgPulsada1;
    var idImgPulsada2;
    var nombreJ1;
    var nombreJ2;
    var jugando;
    var mapFichas;
    var mapFichasMaquina;
    
    this.initialize=function(){
        inicializaMapFichas();
        inicializaMapFichasMaquina();
        inicializaMapImagesAzul();
        inicializaMapImagesRojo();
    };
    
    this.dataContainer=function(contRondas,limRondas,vP1,vP2,turn,modoJuego,chosen1,chosen2,factorAlg,idImg1,idImg2,nJ1,nJ2,playing,mapF,mapFMaq){
        roundsCounter=contRondas;
        roundsLimit=limRondas;
        victoriesP1=vP1;
        victoriesP2=vP2;
        turno=turn;
        modalidadJuego=modoJuego;
        enumChosen1=chosen1;
        enumChosen2=chosen2;
        factorAlgoritmo=factorAlg;
        idImgPulsada1=idImg1;
        idImgPulsada2=idImg2;
        nombreJ1=nJ1;
        nombreJ2=nJ2;
        jugando=playing;
        mapFichas=mapF;
        mapFichasMaquina=mapFMaq;
    };
    
    this.getRoundsCounter=function(){
        return roundsCounter;
    };
    this.getRoundsLimit=function(){
        return roundsLimit;
    };
    this.getVictoriesP1=function(){
        return victoriesP1;
    };
    this.getVictoriesP2=function(){
        return victoriesP2;
    };
    this.isTurno=function(){
        return turno;
    };
    this.getModalidadJuego=function(){
        return modalidadJuego;
    };
    this.getEnumChosen1=function(){
        return enumChosen1;
    };
    this.getEnumChosen2=function(){
        return enumChosen2;
    };
    this.getFactorAlgoritmo=function(){
        return factorAlgoritmo;
    };
    this.getIdImgPulsada1=function(){
        return idImgPulsada1;
    };
    this.getIdImgPulsada2=function(){
        return idImgPulsada2;
    };
    this.getNombreJ1=function(){
        return nombreJ1;
    };
    this.getNombreJ2=function(){
        return nombreJ2;
    };
    this.isJugando=function(){
        return roundsCounter;
    };
    this.getMapFichas=function(){
        return mapFichas;
    };
    this.getMapFichasMaquina=function(){
        return mapFichasMaquina;
    };
    
    this.setRoundsCounter=function(contRondas){
        roundsCounter=contRondas;
    };
    this.setRoundsLimit=function(limRondas){
        roundsLimit=limRondas;
    };
    this.setVictoriesP1=function(vP1){
        victoriesP1=vP1;
    };
    this.setVictoriesP2=function(vP2){
        victoriesP2=vP2;
    };
    this.setTurno=function(turn){
        turno=turn;
    };
    this.setModalidadJuego=function(modoJuego){
        modalidadJuego=modoJuego;
    };
    this.setEnumChosen1=function(chosen1){
        enumChosen1=chosen1; 
    };
    this.setEnumChosen2=function(chosen2){
        enumChosen2=chosen2;
    };
    this.setFactorAlgoritmo=function(factorAlg){
        factorAlgoritmo=factorAlg;
    };
    this.setIdImgPulsada1=function(idImg1){
        idImgPulsada1=idImg1;
    };
    this.setIdImgPulsada2=function(idImg2){
        idImgPulsada2=idImg2;
    };
    this.setNombreJ1=function(nJ1){
        nombreJ1=nJ1;
    };
    this.setNombreJ2=function(nJ2){
        nombreJ2=nJ2;
    };
    this.setJugando=function(playing){
        jugando=playing;
    };
    this.setMapFichas=function(mapF){
        mapFichas=mapF;
    };
    this.setMapFichasMaquina=function(mapFMaq){
        mapFichasMaquina=mapFMaq;
    };
    
    this.inicializaMapFichas=function(){
        var fichas3=new EnumFichas3().getFichas3();
        var fichas5=new EnumFichas5().getFichas5();
        var fichas9=new EnumFichas9().getFichas9();
        mapFichas={
            PAPEL5: fichas5[0],
            PIEDRA5: fichas5[1],
            TIJERA5: fichas5[2],
            SPOCK5: fichas5[3],
            LAGARTO5: fichas5[4],
            PAPEL9: fichas9[0],
            PIEDRA9: fichas9[1],
            TIJERA9: fichas9[2],
            AIRE9: fichas9[3],
            AGUA9: fichas9[4],
            FUEGO9: fichas9[5],
            PISTOLA9: fichas9[6],
            HUMANO9: fichas9[7],
            ESPONJA9: fichas9[8],
            PAPEL3: fichas3[0],
            PIEDRA3: fichas3[1],
            TIJERA3: fichas3[2]
        };
        return mapFichas;
    };
    
    this.inicializaMapFichasMaquina=function(){
        mapFichasMaquina={
            PAPEL:"papelazul",
            PIEDRA:"piedraazul",
            TIJERA:"tijerasazul",
            SPOCK:"spockazul",
            LAGARTO:"lizardazul",
            AIRE:"windazul",
            AGUA:"waterazul",
            PISTOLA:"gunazul",
            FUEGO:"fireazul",
            HUMANO:"humanazul",
            ESPONJA:"spongeazul"
        };
        return mapFichasMaquina;
    };
    
    this.inicializaMapImagesAzul=function(){
        var mapImagesAzul={
            PAPEL5:"papelazul",
            SPOCK5:"spockazul",
            PIEDRA5:"piedraazul",
            TIJERA5:"tijerasazul",
            LAGARTO5:"lizardazul",
            PAPEL9:"windazul",
            AIRE9:"windazul",
            AGUA9:"waterazul",
            PISTOLA9:"gunazul",
            PIEDRA9:"piedraazul",
            FUEGO9:"fireazul",
            TIJERA9:"tijerasazul",
            HUMANO9:"humanazul",
            ESPONJA9:"spongeazul",
            PAPEL3:"papelazul",
            PIEDRA3:"piedraazul",
            TIJERA3:"tijerasazul"
        };
        return mapImagesAzul;
    };
    
    this.inicializaMapImagesRojo=function(){
        var mapImagesAzul={
            PAPEL5:"papelrojo",
            SPOCK5:"spockrojo",
            PIEDRA5:"piedraroja",
            TIJERA5:"tijerasrojo",
            LAGARTO5:"lizardrojo",
            PAPEL9:"windrojo",
            AIRE9:"windrojo",
            AGUA9:"waterrojo",
            PISTOLA9:"gunrojo",
            PIEDRA9:"piedraroja",
            FUEGO9:"firerojo",
            TIJERA9:"tijerasrojo",
            HUMANO9:"humanrojo",
            ESPONJA9:"spongerojo",
            PAPEL3:"papelrojo",
            PIEDRA3:"piedraroja",
            TIJERA3:"tijerasrojo"
        };
        return mapImagesAzul;
    };
    
    this.cambiaTurno=function(){
        turno=!turno;
    };
    
    this.avanzaRonda=function(){
        roundsCounter++;
    };
    
    this.rondasFinalizadas=function(){
        var fin=false;
        if(roundsCounter==roundsLimit){
            fin=true;
        }
        return fin;
    };
    /*
    this.getOrdinalChosen1=function(){
        return enumChosen1.getOrdinal();
    };
    
    this.getOrdinalChosen2=function(){
        return enumChosen2.getOrdinal();
    };
    
    this.setOrdinalChosen1=function(ord){
        enumChosen1.setOrdinal(ord);
    };
    
    this.setOrdinalChosen2=function(ord){
        enumChosen2.setOrdinal(ord);
    };
    
    this.getNameChosen1=function(){
        return enumChosen1.getName();
    };
    
    this.getNameChosen2=function(){
        return enumChosen2.getName();
    };
    
    this.setNameChosen1=function(name){
        enumChosen1.setName(name);
    };
    
    this.setNameChosen2=function(name){
        enumChosen2.setName(name);
    };
    */
    this.sumaVictoriesP1=function(){
        victoriesP1++;
    };
    
    this.sumaVictoriesP2=function(){
        victoriesP2++;
    };
};

File: DesktopApp/src/main/java/modelo/DataContainer.java
package modelo;

import com.mycompany.datapptgame.Fichas3;
import com.mycompany.datapptgame.Fichas5;
import com.mycompany.datapptgame.Fichas9;
import java.util.HashMap;

/**
 * Created by Victor on 29/03/2016.
 */
public class DataContainer {
    private int roundsCounter;
    private int roundsLimit;
    private int victoriesP1;
    private int victoriesP2;
    private boolean turno;
    private int modalidadJuego;
    private Enum chosen1;
    private Enum chosen2;
    private int factorAlgoritmo;
    private String idImagenPulsada1;//En realidad no es el id, sino la ruta, solo se mantiene asi para ver la semejanza con la variable android
    private String idImagenPulsada2;
    private String nombreJ1;
    private String nombreJ2;
    private boolean jugando;
    private HashMap <String,Enum> mapFichas;
    private HashMap <String,String> mapFichasMaquina;

    public DataContainer() {
        inicializaMapFichas();
        inicializaMapFichasMaquina();
    }

    //<editor-fold defaultState="Collapsed" desc="GETTERS Y SETTERS">
    public int getRoundsCounter() {
        return roundsCounter;
    }

    public void setRoundsCounter(int roundsCounter) {
        this.roundsCounter = roundsCounter;
    }

    public int getRoundsLimit() {
        return roundsLimit;
    }

    public void setRoundsLimit(int roundsLimit) {
        this.roundsLimit = roundsLimit;
    }

    public int getVictoriesP1() {
        return victoriesP1;
    }

    public void setVictoriesP1(int victoriesP1) {
        this.victoriesP1 = victoriesP1;
    }

    public int getVictoriesP2() {
        return victoriesP2;
    }

    public void setVictoriesP2(int victoriesP2) {
        this.victoriesP2 = victoriesP2;
    }

    public boolean isTurno() {
        return turno;
    }

    public void setTurno(boolean turno) {
        this.turno = turno;
    }

    public int getModalidadJuego() {
        return modalidadJuego;
    }

    public void setModalidadJuego(int modalidadJuego) {
        this.modalidadJuego = modalidadJuego;
    }

    public Enum getChosen1() {
        return chosen1;
    }

    public void setChosen1(Enum chosen1) {
        this.chosen1 = chosen1;
    }

    public Enum getChosen2() {
        return chosen2;
    }

    public void setChosen2(Enum chosen2) {
        this.chosen2 = chosen2;
    }

    public int getFactorAlgoritmo() {
        return factorAlgoritmo;
    }

    public void setFactorAlgoritmo(int factorAlgoritmo) {
        this.factorAlgoritmo = factorAlgoritmo;
    }

    public String getIdImagenPulsada1() {
        return idImagenPulsada1;
    }

    public void setIdImagenPulsada1(String idImagenPulsada1) {
        this.idImagenPulsada1 = idImagenPulsada1;
    }

    public String getIdImagenPulsada2() {
        return idImagenPulsada2;
    }

    public void setIdImagenPulsada2(String idImagenPulsada2) {
        this.idImagenPulsada2 = idImagenPulsada2;
    }

    public String getNombreJ1() {
        return nombreJ1;
    }

    public void setNombreJ1(String nombreJ1) {
        this.nombreJ1 = nombreJ1;
    }

    public String getNombreJ2() {
        return nombreJ2;
    }

    public void setNombreJ2(String nombreJ2) {
        this.nombreJ2 = nombreJ2;
    }

    public boolean isJugando() {
        return jugando;
    }

    public void setJugando(boolean jugando) {
        this.jugando = jugando;
    }

    public HashMap<String, Enum> getMapFichas() {
        return mapFichas;
    }

    public HashMap<String, String> getMapFichasMaquina() {
        return mapFichasMaquina;
    }
    //</editor-fold>

    //<editor-fold defaultState="Collapsed" desc="GESTION ASIGNACION ENUMS">
    private Enum getCorrectEnum(Enum[] values,String valorBueno){
        Enum devuelve=null;
        boolean sal=false;
        for(int i=0;i<values.length&&!sal;i++){
            if(values[i].name().equalsIgnoreCase(valorBueno)){
                sal=true;
                devuelve=values[i];
            }
        }
        return devuelve;
    }
    //</editor-fold>

    //<editor-fold defaultState="Collapsed" desc="INICIALIZACION HASHMAP">
    private void inicializaMapFichas(){
        mapFichas=new HashMap();
        mapFichas.put("papel5",Fichas5.PAPEL);
        mapFichas.put("spock5",Fichas5.SPOCK);
        mapFichas.put("piedra5",Fichas5.PIEDRA);
        mapFichas.put("tijera5",Fichas5.TIJERA);
        mapFichas.put("lagarto5",Fichas5.LAGARTO);
        mapFichas.put("papel9",Fichas9.PAPEL);
        mapFichas.put("aire9",Fichas9.AIRE);
        mapFichas.put("agua9",Fichas9.AGUA);
        mapFichas.put("pistola9",Fichas9.PISTOLA);
        mapFichas.put("piedra9",Fichas9.PIEDRA);
        mapFichas.put("fuego9",Fichas9.FUEGO);
        mapFichas.put("tijera9",Fichas9.TIJERA);
        mapFichas.put("humano9",Fichas9.HUMANO);
        mapFichas.put("esponja9",Fichas9.ESPONJA);
        mapFichas.put("piedra3",Fichas3.PIEDRA);
        mapFichas.put("papel3",Fichas3.PAPEL);
        mapFichas.put("tijera3",Fichas3.TIJERA);
    }

    private void inicializaMapFichasMaquina() {//Se usa para el setImagenPulsada en caso de jugar contra la maquina, con el .name del enum que sea elegido
        mapFichasMaquina=new HashMap();
        mapFichasMaquina.put("PAPEL","imagenes/papelazul.png");
        mapFichasMaquina.put("SPOCK","imagenes/spockazul.png");
        mapFichasMaquina.put("PIEDRA","imagenes/piedraazul.png");
        mapFichasMaquina.put("TIJERA","imagenes/tijerasazul.png");
        mapFichasMaquina.put("LAGARTO","imagenes/lizardazul.png");
        mapFichasMaquina.put("AIRE","imagenes/windazul.png");
        mapFichasMaquina.put("AGUA","imagenes/waterazul.png");
        mapFichasMaquina.put("PISTOLA","imagenes/gunazul.png");
        mapFichasMaquina.put("FUEGO","imagenes/fireazul.png");
        mapFichasMaquina.put("HUMANO","imagenes/humanazul.png");
        mapFichasMaquina.put("ESPONJA","imagenes/spongeazul.png");
    }

    private HashMap <String,String> inicializaMapImageTagsRojo() {
        HashMap <String,String> mapImageTagsRojo=new HashMap();
        mapImageTagsRojo.put("papel5","imagenes/papelrojo.png");
        mapImageTagsRojo.put("spock5","imagenes/spockrojo.png");
        mapImageTagsRojo.put("piedra5","imagenes/piedraroja.png");
        mapImageTagsRojo.put("tijera5","imagenes/tijerasrojo.png");
        mapImageTagsRojo.put("lagarto5","imagenes/lizardrojo.png");
        mapImageTagsRojo.put("papel9","imagenes/papelrojo.png");
        mapImageTagsRojo.put("aire9","imagenes/windrojo.png");
        mapImageTagsRojo.put("agua9","imagenes/waterrojo.png");
        mapImageTagsRojo.put("pistola9","imagenes/gunrojo.png");
        mapImageTagsRojo.put("piedra9","imagenes/piedraroja.png");
        mapImageTagsRojo.put("fuego9","imagenes/firerojo.png");
        mapImageTagsRojo.put("tijera9","imagenes/tijerasrojo.png");
        mapImageTagsRojo.put("humano9","imagenes/humanrojo.png");
        mapImageTagsRojo.put("esponja9","imagenes/spongerojo.png");
        mapImageTagsRojo.put("piedra3","imagenes/piedraroja.png");
        mapImageTagsRojo.put("papel3","imagenes/papelrojo.png");
        mapImageTagsRojo.put("tijera3","imagenes/tijerasrojo.png");
        return  mapImageTagsRojo;
    }

    private HashMap <String,String> inicializaMapImageTagsAzul() {
        HashMap <String,String> mapImageTagsAzul=new HashMap();
        mapImageTagsAzul.put("papel5","imagenes/papelazul.png");
        mapImageTagsAzul.put("spock5","imagenes/spockazul.png");
        mapImageTagsAzul.put("piedra5","imagenes/piedraazul.png");
        mapImageTagsAzul.put("tijera5","imagenes/tijerasazul.png");
        mapImageTagsAzul.put("lagarto5","imagenes/lizardazul.png");
        mapImageTagsAzul.put("papel9","imagenes/papelazul.png");
        mapImageTagsAzul.put("aire9","imagenes/windazul.png");
        mapImageTagsAzul.put("agua9","imagenes/waterazul.png");
        mapImageTagsAzul.put("pistola9","imagenes/gunazul.png");
        mapImageTagsAzul.put("piedra9","imagenes/piedraazul.png");
        mapImageTagsAzul.put("fuego9","imagenes/fireazul.png");
        mapImageTagsAzul.put("tijera9","imagenes/tijerasazul.png");
        mapImageTagsAzul.put("humano9","imagenes/humanazul.png");
        mapImageTagsAzul.put("esponja9","imagenes/spongeazul.png");
        mapImageTagsAzul.put("piedra3","imagenes/piedraazul.png");
        mapImageTagsAzul.put("papel3","imagenes/papelazul.png");
        mapImageTagsAzul.put("tijera3","imagenes/tijerasazul.png");
        return mapImageTagsAzul;
    }
    //</editor-fold>

    //<editor-fold defaultState="Collapsed" desc="OTROS METODOS">
    public void cambiaTurno(){
        turno=!turno;
    }

    public void sumaVictoriasP1(){
        victoriesP1++;
    }

    public void sumaVictoriasP2(){
        victoriesP2++;
    }

    public int getOrdinalChosen1(){
        return chosen1.ordinal();
    }

    public int getOrdinalChosen2(){
        return chosen2.ordinal();
    }

    public void avanzaRonda(){
        roundsCounter++;
    }

    public boolean rondasFinalizadas(){
        return roundsCounter==roundsLimit;
    }

    public void setValoresIniciales(){
        roundsCounter=0;
        victoriesP1=0;
        victoriesP2=0;
        chosen1=null;
        chosen2=null;
        idImagenPulsada1="";
        idImagenPulsada2="";
    }

    /*public void asignarTagsRojos(AppCompatActivity activity){
        HashMap<String, String> tags=inicializaMapImageTagsRojo();
        for(String key:tags.keySet()){
            activity.findViewById(key).setTag(tags.get(key));
        }
    }

    public void asignarTagsAzules(AppCompatActivity activity){
        HashMap<Integer, Integer> tags=inicializaMapImageTagsAzul();
        for(Integer key:tags.keySet()){
            activity.findViewById(key).setTag(tags.get(key));
        }
    }*/
    //</editor-fold>
}
